core/src/mindustry/editor/MapView.java

```diff
@@ -113,9 +129,25 @@ public void touchUp(InputEvent event, float x, float y, int pointer, KeyCode but
 
                 Point2 p = project(x, y);
 
-                if(tool == EditorTool.line){
-                    ui.editor.resetSaved();
-                    tool.touchedLine(editor, startx, starty, p.x, p.y);
+                switch(tool){
```

Comment:
Do not use standard switch statements anywhere. Either use enhanced switch statements or standard ifs.